--- cos-lite-old.tal
+++ cos-lite-old.tal
@@ -1,3 +1,6 @@
+( collapseOS without filesystem support - for old uxnasm commit 8d14b4bbc6f1e50f313293ba301268fbc7d9090e
+  for https://aduros.com/webuxn/ )
+
 ( constants )
 %SYSVARS+#18 { #fe18 }
 %RS_ADDR { #ff00 }
@@ -9,6 +12,9 @@
 %RTN { JMP2r }
 %HALF2 { DUP2 #8000 AND2 #01 SFT2 ORA2 } ( signed division by 2 )
 
+%INC2 { #0001 ADD2 }
+%INC { #01 ADD }
+
 ( zero page )
 |00 @ZP [ &psp $2 &rsp $2 &ir $2 &pc $2 &curX $1 &curY $1 &tmpX $1 &tmpY $1 &tmp1 $2 &tmp2 $2 ]
 
@@ -21,7 +27,7 @@
 ( devices )
 |00 @System [ &vector $2 &wst $1 &rst $1 &pad $4 &r $2 &g $2 &b $2 &debug $1 &halt $1 ]
 |10 @Console [ &vector $2 &read $1 &pad $5 &write $1 &error $1 ]
-|20 @Screen [ &vector $2 &width $2 &height $2 &pad $2 &x $2 &y $2 &addr $2 &pixel $1 &sprite $1 ]
+|20 @Screen [ &vector $2 &width $2 &height $2 &pad $2 &x $2 &y $2 &addr $2 &color $1 ]
 |80 @Controller [ &vector $2 &button $1 &key $1 ]
 
 ( stable ABI )
@@ -213,20 +219,20 @@
 	#00 .ZP/tmpY LDZ INC2 #0008 MUL2 .Screen/y DEO2
 	#00 .ZP/tmpX LDZ INC2 #0008 MUL2 .Screen/x DEO2
 	#00 .ZP/tmp2 LDZ #20 SUB #0007 MUL2 .ZP/tmp1 LDZ2 ADD2 .Screen/addr DEO2
-	#49 .Screen/sprite DEO
+	#29 .Screen/color DEO
 	;CursorSprite .Screen/addr DEO2
-	#4a .Screen/sprite DEO
+	#32 .Screen/color DEO
 	,&done JMP
 	&cursor
 	;CursorSprite .Screen/addr DEO2
 	#00 .ZP/curY LDZ INC2 #0008 MUL2 .Screen/y DEO2
 	#00 .ZP/curX LDZ INC2 #0008 MUL2 .Screen/x DEO2
-	#4a .Screen/sprite DEO
+	#32 .Screen/color DEO
 	.ZP/tmpX LDZ .ZP/curX STZ
 	.ZP/tmpY LDZ .ZP/curY STZ
 	#00 .ZP/curY LDZ INC2 #0008 MUL2 .Screen/y DEO2
 	#00 .ZP/curX LDZ INC2 #0008 MUL2 .Screen/x DEO2
-	#4f .Screen/sprite DEO
+	#31 .Screen/color DEO
 
 	&done
 	;handler/next JMP2
@@ -348,8 +354,6 @@
 @op-uxnDEO2
 	;ppop JSR2 ;ppop JSR2 SWP2 SWP POP DEO2
 	;handler/next JMP2
-( collapseOS without filesystem support )
-~common.tal
 
 @blkinit RTN
 
@@ -383,12 +387,12 @@
 	OVR INC #00 SWP #0008 MUL2 .Screen/y DEO2
 	&cellloop
 	DUP INC #00 SWP #0008 MUL2 .Screen/x DEO2
-	#49 .Screen/sprite DEO
+	#29 .Screen/color DEO
 	INC DUP #50 LTH ,&cellloop JCN
 	POP INC #00 OVR #19 LTH ,&lineloop JCN
 	POP2
 	( dummy device-out as replacement for loading ROM )
-	#0000 .System/vector
+	;key-handler .Controller/vector
 	( go to main program )
 	;main JMP2
 
